src/test/java/com/commercetools/sync/customobjects/utils/CustomObjectSyncUtilsTest.java

```diff
@@ -0,0 +1,249 @@
+                .put("username", "Peter");
+
+        JsonNode oldJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", oldNestedJson);
+
+        JsonNode newJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", newNestedJson);
+
+
+        newCustomObjectdraft = CustomObjectDraft.ofUnversionedUpsert(container, key, newJsonNode);
+        oldCustomObject = mock(CustomObject.class);
+        when(oldCustomObject.getValue()).thenReturn(oldJsonNode);
+        when(oldCustomObject.getContainer()).thenReturn(container);
+        when(oldCustomObject.getKey()).thenReturn(key);
+
+        assertThat(oldJsonNode.toString()).isNotEqualTo(newJsonNode.toString());
+        assertThat(CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObjectdraft)).isTrue();
+    }
+
+    @Test
+    void hasIdenticalValue_WithDifferentNestedJsonNode_ShouldNotBeIdentical() {
+
+        JsonNode oldNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("username", "Peter")
+                .put("userId", "123-456-789");
+        JsonNode newNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("userId", "129-382-189")
+                .put("username", "Peter");
+
+        JsonNode oldJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", oldNestedJson);
+
+        JsonNode newJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", newNestedJson);
+
+
+        newCustomObjectdraft = CustomObjectDraft.ofUnversionedUpsert(container, key, newJsonNode);
+        oldCustomObject = mock(CustomObject.class);
+        when(oldCustomObject.getValue()).thenReturn(oldJsonNode);
+        when(oldCustomObject.getContainer()).thenReturn(container);
+        when(oldCustomObject.getKey()).thenReturn(key);
+
+        assertThat(oldJsonNode.toString()).isNotEqualTo(newJsonNode.toString());
+        assertThat(CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObjectdraft)).isFalse();
+    }
+
+    @Test
+    void batchElements_WithValidSize_ShouldReturnCorrectBatches() {
+        final int numberOfCustomObjectDrafts = 160;
+        final int batchSize = 10;
+        final ArrayList<CustomObjectDraft<JsonNode>> customObjectDrafts = new ArrayList<>();
+
+        for (int i = 0; i < numberOfCustomObjectDrafts; i++) {
+            customObjectDrafts.add(CustomObjectDraft
+                    .ofUnversionedUpsert("container_" + i, "key_" + i,
+                            JsonNodeFactory.instance.objectNode().put("field", "field_" + i)));
+        }
+        final List<List<CustomObjectDraft<JsonNode>>> batches = batchElements(customObjectDrafts, 10);
+        assertThat(batches.size()).isEqualTo(numberOfCustomObjectDrafts / batchSize);
+    }
+
+    @Test
+    void batchElements_WithNegativeSize_ShouldReturnNoBatches() {
+        final int numberOfCustomObjectDrafts = 160;
```

Comment:
Remove this unused var.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/561/commits/cce740bb6b8df08afa4c1a69ab4bf71889bb7762